nrcatalogtools/lvc.py

```diff
@@ -224,3 +224,462 @@ def get_strain_from_lvcnr_file(path_to_file,
+                          The components of a three-vector
+
+    Returns
+    -------
+    vec_x, vec_y, vec_z : float
+                          The normalized components of the three-vector.
+    '''
+
+    norm_vec = Norm(vec_x, vec_y, vec_z)
+
+    return vec_x/norm_vec, vec_y/norm_vec, vec_z/norm_vec\
+
+def CrossProduct(vec1_x, vec1_y, vec1_z, vec2_x, vec2_y, vec2_z):
+    ''' Compute the cross product of the two input vectors
+
+    Parameters
+    ----------
+    vec1_x, vec1_y, vec1_z, vec2_x, vec2_y, vec2_z  : float
+                                                      The components of the two three-vectors :math:`\vec{v_1}, \vec{v_2}`
+
+    Returns
+    -------
+    vec1_cross_vec2_x, vec1_cross_vec2_y, vec1_cross_vec2_z : float
+                                                              The cross product :math:`\vec{v_1} \times \vec{v_2}`
+    '''
+
+    vec1_cross_vec2_x = vec1_y * vec2_z - vec1_z * vec2_y
+    vec1_cross_vec2_y = vec1_z * vec2_x - vec1_x * vec2_z
+    vec1_cross_vec2_z = vec1_x * vec2_y - vec1_y * vec2_x
+
+    return vec1_cross_vec2_x, vec1_cross_vec2_y, vec1_cross_vec2_z
+
+
+def CheckInterpReq(H5File, ref_time):
+    ''' Check if the required reference time is different from
+    the available reference time in the NR HDF5 file
+
+    Parameters
+    ----------
+    H5File : file object
+                The waveform h5 file handle.
+    ref_time : float
+               Reference time.
+
+    Returns
+    -------
+    Interp : bool
+             Whether interpolation across time is required.
+    avail_ref_time: float
+                    The ref_time available in the NR HDF5 file.
+    '''
+
+    avail_ref_time = H5File.attrs('ref_time')
+
+    if abs(avail_ref_time-ref_time)<1e-5:
+        return False, avail_ref_time
+    else:
+        return True, avail_ref_time
+
+
+def GetRefFreqFromRefTime(H5File, ref_time):
+    ''' Get the reference frequency from reference time 
+    
+    Parameters
+    ----------
+    H5File : file object
+             The waveform h5 file handle.
+    ref_time : float
+               Reference time.
+    Returns
+    -------
+    fRef : float
+           Reference frequency.
+    '''
+    
+    time, freq = H5File.attrs['Omega-vs-time']
+    
+    from scipy.interpolate import interp1d
+    
+    RefFreq = interp1d(time, freq, kind='cubic')[ref_time]
+    
+    return RefFreq
+
+def GetRefTimeFromRefFreq(H5File, ref_freq):
+    ''' Get the reference time from reference frequency 
+    
+    Parameters
+    ----------
+    H5File : file object
+             The waveform h5 file handle.
+    ref_freq : float
+               The reference frequency.
+    Returns
+    -------
+    fTime : float
+           Reference time.
+    '''
+    from scipy.interpolate import interp1d
+    time, freq = H5File.attrs['Omega-vs-time']
+    
+    
+    
+    RefTime = interp1d(freq, time, kind='cubic')[ref_freq]
+    
+    return RefTime
+
+def CheckNRAttrs(H5File, ReqAttrs):
```

Comment:
Would it be useful to have a reasonable default for `ReqAttrs` here given the LVCNR format is, per-se, fixed?
For e.g. one could gather all the attributes being read in from H5 files here in `lvc.py` and collect them as a list here..